src/main/java/codeu/controller/FileUploadServlet.java

```diff
@@ -0,0 +1,41 @@
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.StandardCopyOption;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.Part;
+
+public class FileUploadServlet extends HttpServlet {
+
+	@Override
+	public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
+
+		//get the file chosen by the user
+		Part filePart = request.getPart("fileToUpload");
+		
+		//get the InputStream to store the file somewhere
+	    InputStream fileInputStream = filePart.getInputStream();
+	    
+	    //for example, you can copy the uploaded file to the server
```

Comment:
Rather than doing everything below this part, we want to use the media and media store class that I created to save the files. This would be the code necessary to accomplish that task:

      User user = userStore.getUser(mediaOwner);
      if(inputStream!=null) {
        BufferedImage content = ImageIO.read(inputStream);
        String contentType = filePart.getContentType();
        Media media = new Media(UUID.randomUUID(), user.getId(), 
          mediaTitle, Instant.now(), content, contentType);
        if(mediaStore.getProfilePicture(user.getName())!=null) {
          mediaStore.getProfilePicture(user.getName()).setIsProfilePicture(false);
        }
        mediaStore.setProfilePicture(user.getName(), media);
        mediaStore.addMedia(media);
      }
        
      response.sendRedirect("/profile/"+user.getName());

Reply:
I should also note that the above code will work once my pull request goes through.